native-audio/app/src/main/java/com/example/nativeaudio/NativeAudio.java

```diff
@@ -308,6 +336,35 @@ protected void onDestroy()
         shutdown();
         super.onDestroy();
     }
+    private void requestResourcePermission() {
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.RECORD_AUDIO) !=
+                PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[]{
+                            Manifest.permission.RECORD_AUDIO,
+                            Manifest.permission.MODIFY_AUDIO_SETTINGS,
+                            Manifest.permission.INTERNET },
+                    AUDIO_ECHO_REQUEST);
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
+                                           @NonNull int[] grantResults) {
+
+        /*
+         * if any permission failed, the sample could not play
+         */
+        if (AUDIO_ECHO_REQUEST != 0) {
```

Comment:
wouldn't this be always true? (because `private static final int AUDIO_ECHO_REQUEST = 0;`)